resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -292,6 +298,19 @@ int dnsRedirectPort(@SuppressWarnings("unused") InetAddress server) {
         return DNS_PORT;
     }
 
+    final DnsQueryLifecycleObserverFactory dnsQueryLifecycleObserverFactory() {
+        return dnsQueryLifecycleObserverFactory;
+    }
+
+    /**
+     * Provides the opportunity to sort the name servers before following a redirected DNS query.
+     * @param nameServers The addresses of the DNS servers which are used in the event of a redirect.
+     * @return A {@link DnsServerAddressStream} which will be used to follow the DNS redirect.
+     */
+    public DnsServerAddressStream uncachedRedirectDnsServerStream(List<InetSocketAddress> nameServers) {
+        return DnsServerAddresses.sequential(nameServers).stream();
```

Comment:
The intention is to have some way the user can impact the order in which DNS servers are queried when we get a redirect response that is not in the cache. I'm open to alternatives if you have a better approach for this.